### PR TITLE
Resource template for BGP neighbor encryption_password arg

### PR DESCRIPTION
**Pull request by sc68cal**
Depends-on: https://github.com/ansible/ansible-zuul-jobs/pull/1601

The `bgp_global` module has an attribute in BGP neighbors for setting the password, however the code was not implemented.

### CODE REVIEW

**Comment by sc68cal** [comment]
TODO: add unit tests

In the meantime we have used this in our production environment and it is allowing us to move forward with our deployment

File: changelogs/fragments/bgp_global_neighbor_password.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+-minor_changes:
+-  - Add support for setting encryption_password for BGP neighbors in bgp_global module
```
Comment (Qalthos) [inline changelogs/fragments/bgp_global_neighbor_password.yml:3]:
```suggestion
minor_changes:
  - Add support for setting encryption_password for BGP neighbors in bgp_global module
```

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: changelogs/fragments/bgp_global_neighbor_password.yml (hunk 1 above)
Comment (sc68cal) [reply changelogs/fragments/bgp_global_neighbor_password.yml:3]: whoops. That's what I get for looking at `git log -p` for an example of the format

**Comment by sc68cal** [comment]
So, I took a look at where to put testing for this, and I got a little overwhelmed. There's both an `integration` and `unit` directory in tests, and I am a little unsure if I am supposed to add it to both?

**Comment by sc68cal** [comment]
Just an update on this, having to juggle some other priorities, so I'll try and complete this next week 🤞

**Comment by sc68cal** [comment]
Instead of figuring out how to create a whole new set of fixtures and tests for just the password, I decided to add to the existing tests for `bgp_global` and set `peer2` to have a password.

I did not rebase this PR but instead used a merge strategy, since I don't know the policy of this repo when it comes to rebase/rewriting history. I would be fine with this being squashed down to just two commits, one where I implement the feature and tests and then one commit where @Qalthos fixed my YAML syntax so they still receive credit for their work.

**Comment by Qalthos** [comment]
recheck

**Comment by Qalthos** [comment]
recheck

**Comment by sc68cal** [comment]
recheck

**Comment by Qalthos** [comment]
recheck

**Review by rohitthakur2590 — APPROVED** [review]

**Comment by sc68cal** [comment]
https://github.com/ansible-collections/arista.eos/runs/7653633876?check_suite_focus=true#step:6:131 

Looks like a connection to Ansible galaxy timed out, recheck?

**Comment by rohitthakur2590** [comment]
recheck

**Pull request merged**